pages/api/stats.ts

```diff
@@ -0,0 +1,34 @@
+import { withApiMethods } from "../../lib/withApiMethods";
+import { prisma } from "../../lib/db";
+import { generateStats } from "../../lib/stats/stats";
+
+export default withApiMethods({
+  GET: async (req, res) => {
+    const stats = await prisma.stat.findMany();
+
+    if (!stats.length || stats[0]?.updatedAt < new Date(Date.now() - 1000 * 10)) {
```

Comment:
I'd move the part responsible for calculating the expiry of stats to another file/function/constant